lib/edge_rider/collect_ids.rb

```diff
@@ -7,12 +7,12 @@ module Array
 
       def collect_ids
         collect do |obj|
-          case obj
-            when Integer then obj
-            when ActiveRecord::Base then obj.id
-            else raise Uncollectable, "Cannot collect an id from #{obj.inspect}"
+          if obj.respond_to?(:collect_ids)
+            obj.collect_ids
+          else
+            raise Uncollectable, "Cannot collect an id from #{obj.inspect}"
           end
-        end
+        end.flatten
```

Comment:
Please add a spec for your use of `flatten`.

Reply:
We have reconsidered this: could you make this less smart? There should not be a need to support nested Arrays, so please remove the recursion and add `when String` instead.

I'd like to avoid performance issues when calling it e.g. on deeply nested arrays from params.